File: app/src/main/java/com/example/assignment1/Model/MovieList.java
package com.example.assignment1.Model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class MovieList {

    @SerializedName("page")
    @Expose
    private Integer page;
    @SerializedName("results")
    @Expose
    private final List<PopularMovie> popularMovies = new ArrayList<PopularMovie>();
    @SerializedName("total_results")
    @Expose
    private Integer totalResults;
    @SerializedName("total_pages")
    @Expose
    private Integer totalPages;


    public List<PopularMovie> getResults() {
        return popularMovies;
    }


}

File: app/src/main/java/com/example/assignment1/Model/MovieList.kt
package com.example.assignment1.Model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class MovieList {
    @SerializedName("page")
    @Expose
    private val page: Int? = null

    @SerializedName("results")
    @Expose
    val results: List<PopularMovie> = ArrayList()

    @SerializedName("total_results")
    @Expose
    private val totalResults: Int? = null

    @SerializedName("total_pages")
    @Expose
    private val totalPages: Int? = null
}